bindings/src/main/java/io/pravega/storage/azure/AzureChunkStorage.java

```diff
@@ -0,0 +1,210 @@
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenRead");
+        }
+        return ChunkHandle.readHandle(chunkName);
+    }
+
+    @Override
+    protected ChunkHandle doOpenWrite(String chunkName) throws ChunkStorageException {
+        if (!checkExists(chunkName)) {
+            throw new ChunkNotFoundException(chunkName, "doOpenWrite");
+        }
+        return new ChunkHandle(chunkName, false);
+    }
+
+    @Override
+    protected int doRead(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) throws ChunkStorageException {
+        try  {
+            try (val inputStream = client.getInputStream(getObjectPath(handle.getChunkName()), fromOffset, length)) {
+                val retValue =  StreamHelpers.readAll(inputStream, buffer, bufferOffset, length);
+                if (retValue == 0) {
+                    val blobProperties = client.getBlobProperties(getObjectPath(handle.getChunkName()));
+                    if (blobProperties.getBlobSize() <= fromOffset || blobProperties.getBlobSize() <= fromOffset + length) {
+                        throw new IllegalArgumentException();
+                    }
+                }
+                return retValue;
+            }
+        } catch (IllegalArgumentException e) {
+                throw e;
+        } catch (Exception e) {
+            throw convertException(handle.getChunkName(), "doRead", e);
+        }
+    }
+
+    @Override
+    protected int doWrite(ChunkHandle handle, long offset, int length, InputStream data) throws ChunkStorageException {
+        try {
+            val objectPath = getObjectPath(handle.getChunkName());
+            val metadata = client.getBlobProperties(objectPath);
+            if (metadata.getBlobSize() != offset) {
+                throw new InvalidOffsetException(handle.getChunkName(), metadata.getBlobSize(), offset, "doWrite");
+            }
+            client.appendBlock(objectPath, offset, length, new BufferedInputStream(data));
+            return length;
+        } catch (Exception e) {
+            throw convertException(handle.getChunkName(), "doWrite", e);
+        }
+    }
+
+    @Override
+    protected int doConcat(ConcatArgument[] chunks) throws UnsupportedOperationException {
+        throw new UnsupportedOperationException("AzureChunkStorage does not support concat operation.");
+    }
+
+    @Override
+    protected void doSetReadOnly(ChunkHandle handle, boolean isReadOnly) throws UnsupportedOperationException, ChunkStorageException {
+        throw new UnsupportedOperationException("AzureChunkStorage does not support setReadOnly operation.");
+    }
+
+    @Override
+    @SneakyThrows
+    public void close() {
+        if (shouldClose && !this.closed.getAndSet(true)) {
+            this.client.close();
+        }
+        super.close();
+    }
+
+    private ChunkStorageException convertException(String chunkName, String message, Exception e) {
```

Comment:
Add method description

Reply:
The method is private and hence restricted to that class. So, I believe the method is self-explanatory with the exceptions listed in it with regards to the BlobErrorCodes. But if you suggest so I'll provide link to the class from where we relating ChunkStorageExceptions and the errorCodes for those respective exceptions.